WoWDatabaseEditor.Common/WDE.DbcStore/Providers/DBDProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using DBCD.Providers;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace WDE.DbcStore.Providers
+{
+    public class DBDProvider : IDBDProvider
+    {
+        private static Uri BaseURI = new Uri("https://raw.githubusercontent.com/wowdev/WoWDBDefs/master/definitions/");
+        private static string CachePath = "dbdCache/";
+        private HttpClient client = new HttpClient();
+
+        public DBDProvider()
+        {
+            if (!Directory.Exists(CachePath))
+                Directory.CreateDirectory(CachePath);
+
+            client.BaseAddress = BaseURI;
+        }
+
+        public Stream StreamForTableName(string tableName, string build = null)
+        {
+            string dbdName = Path.GetFileName(tableName).Replace(".db2", ".dbd");
+
+            if (!File.Exists($"{CachePath}/{dbdName}") || (DateTime.Now - File.GetLastWriteTime($"{CachePath}/{dbdName}")).TotalHours > 24)
+            {
+                try
+                {
+                    var bytes = client.GetByteArrayAsync(dbdName).Result;
```

Comment:
I know this comes from DBCD code, but it doesn't look good to me, even though this code will be run on a thread, I don't like using .Result of await task.

You can read more here: https://stackoverflow.com/questions/14435520/why-use-httpclient-for-synchronous-connection/14435574.

And the safe solution is to use sync call instead of async: https://stackoverflow.com/a/66656039/1616645